### PR TITLE
[38_8] convert: use find-binary-convert and tuning on windows

### PR DESCRIPTION
**Pull request by darcy-shen**
## What
+ Use `find-binary-convert` for imagemagick command line finding

This pull request is only for `find-binary-convert` adoption. Gif image rendering still needs to be improved.

## Why
Unify it.

## How to test your changes?
+ [x] macOS
+ [ ] Windows
+ [ ] Linux

On windows:
```
scoop install main/imagemagick
```

### Test gif animation
Just open `TeXmacs/tests/38_8.tm`.

Tested on macOS, should work fine on linux.

Insert a gif image via Animation, see http://forum.texmacs.cn/t/video-inclusion/683
```
TeXmacs] debug-io, call: /opt/homebrew/bin/convert +adjoin -coalesce /Users/da/git/mogan/50e54075874a46b0c8945d2384a1baf8.gif /tmp/.lolly/92788/06FC3513_BACD_4110_89F7_27321B2E9A30_%04d.gif
```

### Test webp conversion
Just open `TeXmacs/tests/38_8.tm`

### CODE REVIEW

**Comment by jingkaimori** [comment]
`find-binary-convert` cannot find `convert` on windows. `convert` is installed from [imagemagick](https://imagemagick.org/script/download.php#windows) with option "install legacy component (convert)" enabled.

[convert定位测试.zip](https://github.com/XmacsLabs/mogan/files/14662481/convert.zip)

**Comment by Oyyko** [comment]
> `find-binary-convert` cannot find `convert` on windows. `convert` is installed from [imagemagick](https://imagemagick.org/script/download.php#windows) with option "install legacy component (convert)" enabled.
> 
> [convert定位测试.zip](https://github.com/XmacsLabs/mogan/files/14662481/convert.zip)


Yes, I think it is possible if the user does not have ImageMagick installed. We should consider this situation.

**Comment by darcy-shen** [comment]
ImageMagick这个命令本来就没有为Windows适配吧，所以这个pr也不做相关适配。

**Comment by darcy-shen** [comment]
> option "install legacy component (convert)" enabled

统一用scoop就不存在这个问题了。现在的问题是 lolly::system::call 调用命令会失败。

**Comment by darcy-shen** [comment]
> 现在的问题是 lolly::system::call 调用命令会失败。

这个也没问题了，现在唯一的问题是，我发现我从Windows的elvish里面启动会出问题。不过这个属于开发者的问题

**Comment by darcy-shen** [comment]
这个pr本身是完整的，由于 @jingkaimori 反馈了，设置PATH之后，目前find-binary-xxx的实现有问题，故而先解决这个PATH下exe的查找问题，再继续完成这个pr

File: TeXmacs/plugins/binary/progs/binary/identify.scm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+;; DESCRIPTION : Imagemagick Binary plugin: identify
+;; COPYRIGHT   : (C) 2024  Darcy Shen
+;;
+;; This software falls under the GNU general public license version 3 or later.
+;; It comes WITHOUT ANY WARRANTY WHATSOEVER. For details, see the file LICENSE
+;; in the root directory or <http://www.gnu.org/licenses/gpl-3.0.html>.
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(texmacs-module (binary identify))
```
Comment (jingkaimori) [inline TeXmacs/plugins/binary/progs/binary/identify.scm:14]: missing `(:use (binary common))`

**Review by jingkaimori — APPROVED** [review]
LGTM

**Pull request merged**